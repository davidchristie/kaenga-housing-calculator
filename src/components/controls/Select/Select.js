import PropTypes from 'prop-types'
import React, { Component } from 'react'
import {
  ControlLabel,
  FormControl,
  FormGroup,
  HelpBlock
} from 'react-bootstrap'

class Select extends Component {
  render () {
    const {
      input: { onChange, value },
      label,
      meta: {
        error
      },
      multiple
    } = this.props
    return (
      <FormGroup>
        {label ? <ControlLabel>{label}</ControlLabel> : null}
        <FormControl
          componentClass='select'
          multiple={multiple}
          onChange={onChange}
          value={value}
        >
          {this.props.children}
        </FormControl>
        {error && <HelpBlock>{error}</HelpBlock>}
      </FormGroup>
    )
  }
}

Select.defaultProps = {
  input: {},
  meta: {}
}

Select.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired,
  label: PropTypes.string,
  multiple: PropTypes.bool
}

export default Select
