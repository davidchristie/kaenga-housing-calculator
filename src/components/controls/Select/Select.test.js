import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import Select from './Select'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Select />)).toMatchSnapshot()
})
