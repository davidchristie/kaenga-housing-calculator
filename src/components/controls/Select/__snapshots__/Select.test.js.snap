exports[`test matches snapshot 1`] = `
<FormGroup
  bsClass="form-group">
  <FormControl
    bsClass="form-control"
    componentClass="select"
    multiple={undefined}
    onChange={undefined}
    value={undefined} />
</FormGroup>
`;
